tests/functional/backend/wmg/test_wmg_api.py

```diff
@@ -0,0 +1,113 @@
+        Returns in less than 10 seconds
+        """
+        MAX_RESPONSE_TIME_SECONDS = 10
+        headers = {"Content-Type": "application/json"}
+        data = {
+            "filter": {
+                "dataset_ids": [],
+                "disease_ontology_term_ids": ["MONDO:0005812", "MONDO:0100096", "PATO:0000461"],
+                "ethnicity_ontology_term_ids": ["HANCESTRO:0010", "HANCESTRO:0014"],
+                "gene_ontology_term_ids": list(genes_20_count.keys()),
+                "organism_ontology_term_id": "NCBITaxon:9606",
+                "sex_ontology_term_ids": ["PATO:0000383"],
+                "tissue_ontology_term_ids": ["UBERON:0000178"],
+            },  # blood (more than 50 cell types)
+            "include_filter_dims": True,
+            "snapshot_id": self.data["snapshot_id"],
+        }
+        with self.subTest("Run once"):
+            res = requests.post(f"{self.api}/query", data=json.dumps(data), headers=headers)
+            self.assertGreaterEqual(MAX_RESPONSE_TIME_SECONDS, res.elapsed.seconds)
+            self.assertEqual(res.status_code, requests.codes.ok)
+            self.assertGreater(len(res.content), 10)
+        with self.subTest("Run 10 times"):
+            with self.subTest("Test 10 times"):
+
+                def make_request():
+                    return requests.post(f"{self.api}/query", data=json.dumps(data), headers=headers)
+
+                seconds_for_10_runs = timeit.timeit(setup="", stmt=make_request, number=10)
+                self.assertGreater(MAX_RESPONSE_TIME_SECONDS * 10, seconds_for_10_runs)
+
+    def test_secondary_filters_extreme_case(self):
+        """
+        4 tissues w/largest cell type counts, 400 genes, no secondary filtering
+        Returns in less than 15 seconds
+        """
+        MAX_RESPONSE_TIME_SECONDS = 15
+        headers = {"Content-Type": "application/json"}
+        data = {
+            "filter": {
+                "dataset_ids": [],
+                "disease_ontology_term_ids": [],
+                "ethnicity_ontology_term_ids": [],
+                "gene_ontology_term_ids": list(genes_400_count.keys()),
+                "organism_ontology_term_id": "NCBITaxon:9606",
+                "sex_ontology_term_ids": [],
+                "tissue_ontology_term_ids": [
+                    "UBERON:0000178",
+                    "UBERON:0002048",
+                    "UBERON:0000029",
+                    "UBERON:0000970",
+                    "UBERON:0000362",
+                ],
+            },
+            "include_filter_dims": True,
+            "snapshot_id": self.data["snapshot_id"],
+        }
+        with self.subTest("Test once"):
+            res = requests.post(f"{self.api}/query", data=json.dumps(data), headers=headers)
+            self.assertGreater(MAX_RESPONSE_TIME_SECONDS, res.elapsed.seconds)
+            self.assertEqual(res.status_code, requests.codes.ok)
+            self.assertGreater(len(res.content), 10)
```

Comment:
Is this single-call test still needed if we're doing an avg-of-10 timing test below? The motivation for validating avg performance instead of single-call performance was to avoid test failures due to "bad luck" (e.g. cold cache). 

Also, if the  MAX_RESPONSE_TIME_SECONDS is currently set to a conservative (higher) value than needed to account for "bad luck" response times, maybe we can lower it if we're just doing a 10-call average.